src/apps/plutus/arbitrum/plutus.balance-fetcher.ts

```diff
@@ -0,0 +1,149 @@
+import { Inject } from '@nestjs/common';
+
+import { IAppToolkit, APP_TOOLKIT } from '~app-toolkit/app-toolkit.interface';
+import { Register } from '~app-toolkit/decorators';
+import { presentBalanceFetcherResponse } from '~app-toolkit/helpers/presentation/balance-fetcher-response.present';
+import { BalanceFetcher } from '~balance/balance-fetcher.interface';
+import { Network } from '~types/network.interface';
+
+import {
+  PlutusContractFactory,
+  PlsPlutusChef,
+  PlsDpxPlutusChef,
+  PlsJonesPlutusChef,
+  PlutusEpochStaking,
+} from '../contracts';
+import { PLUTUS_DEFINITION } from '../plutus.definition';
+
+const appId = PLUTUS_DEFINITION.id;
+const network = Network.ARBITRUM_MAINNET;
+
+@Register.BalanceFetcher(PLUTUS_DEFINITION.id, network)
+export class ArbitrumPlutusBalanceFetcher implements BalanceFetcher {
+  constructor(
+    @Inject(APP_TOOLKIT) private readonly appToolkit: IAppToolkit,
+    @Inject(PlutusContractFactory) private readonly contractFactory: PlutusContractFactory,
+  ) {}
+
+  async getTokenBalances(address: string) {
+    return this.appToolkit.helpers.tokenBalanceHelper.getTokenBalances({
+      address,
+      network,
+      appId,
+      groupId: PLUTUS_DEFINITION.groups.ve.id,
+    });
+  }
+
+  async getLockedBalances(address: string) {
+    return this.appToolkit.helpers.singleStakingContractPositionBalanceHelper.getBalances<PlutusEpochStaking>({
+      address,
+      network,
+      appId,
+      groupId: PLUTUS_DEFINITION.groups.lock.id,
+      resolveContract: ({ address, network }) => this.contractFactory.plutusEpochStaking({ address, network }),
+      resolveStakedTokenBalance: ({ contract, address, multicall }) =>
+        multicall
+          .wrap(contract)
+          .stakedDetails(address)
+          .then(details => details.amount),
+      resolveRewardTokenBalances: () => [], // TODO: implement
+    });
+  }
+
+  async getStakedDPXBalances(address: string) {
+    return this.appToolkit.helpers.singleStakingContractPositionBalanceHelper.getBalances<PlsDpxPlutusChef>({
```

Comment:
Use the master chef balance helper:
```suggestion
    return this.appToolkit.helpers.masterChefContractPositionHelper.getContractPositions<PlsPlutusChef>({
```

See examples in `src/apps/abracadabra/arbitrum/abracadabra.farm.contract-position-fetcher.ts`